pyrtl/memory.py

```diff
@@ -259,10 +259,7 @@ def _build_read_port(self, addr):
     def _assignment(self, item, val, is_conditional):
         from .conditional import _build
 
-        item = as_wires(item, bitwidth=self.addrwidth, truncating=False)
-        if len(item) > self.addrwidth:
-            raise PyrtlError('error, the wire indexing the memory bitwidth > addrwidth')
-        addr = item
+        addr = as_wires(item, bitwidth=self.addrwidth, truncating=False)
```

Comment:
Do we need to call `as_wires` here? It seems like that already happens in `__getitem__`. If this is needed, consider adding a comment explaining why, otherwise we can remove it?